x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportPutFeatureIndexBuilderJobAction.java

```diff
@@ -63,7 +63,7 @@ protected String executor() {
     protected void masterOperation(Request request, ClusterState clusterState, ActionListener<Response> listener) throws Exception {
 
         if (!licenseState.isMachineLearningAllowed()) {
-            listener.onFailure(LicenseUtils.newComplianceException(XPackField.FIB));
```

Comment:
Why are we still checking `!licenseState.isMachineLearningAllowed()` in the line before this? We should be uniform in what license we require to use this feature, no?

Reply:
good catch! should be `isDataFrameAllowed()`